[terra-table] Uplift Table components, create HTML Table (#2615)
cerner/terra-core: packages/terra-table/src/terra-dev-site/doc/table/guides.91/SingleSelectTable.1.doc.mdx

import SingleSelectTable from 'terra-table/lib/terra-dev-site/doc/guides/SingleSelectTable?dev-site-example';

# Terra Table - Implementing a Single Select Table

> The terra-table implementation requires controlled state if selections are required. As a result selections are applied at child row generation from the consuming component's state. The following guide show you how to implement that state within a single row selection variant of table.

Reviewer: ```suggestion
This guide will show you how to implement row selection state within a single row selection variant of table. To enable row selection, terra-table requires selection state to be managed by the consuming component. As a result selections from the consuming components state are applied at generation of the table rows.
```

Author: Updated.

## State Management
The state of selection needs to be managed for the table in an ancestor component. In this example we are going to use a unique key, but the type of state used is open to the individual implementation.

 First defaulting our state to a null value with the useState hook.
```diff
+ const [selectedKey, setSelectedKey] = useState([]);
```
Next creating an event handler callback method to pass to the table row's "onToggle" prop. The "onToggle" will return the metaData prop passed it each row.
```diff
+  const handleRowToggle = (event, metaData) => {

+  };
```
As a precaution we can prevent default on the event, in case the table has an ancestor with a listener. This also prevents the browser from auto page scrolling when we are intending to make a selection with the space bar.
```diff
  const handleRowToggle = (event, metaData) => {
+   event.preventDefault();
  };
```

> A single select table normally doesn't allow deselection, so we'll be using it in our example, but if deselection is desired we could adjust this method. So we compare the key to the current one in state, and if they aren't the same we set this as our state.

Reviewer: ```suggestion
Compare the new key to the current key in state and if they aren't the same, set the new key in state.
Single select tables normally don't allow deselection, so neither will this example. If deselection is desired adjust this method to toggle selection.
```

Author: Updated.

```diff
  const handleRowToggle = (event, metaData) => {
    event.preventDefault();
+   if (selectedKey !== metaData.key) {
+     setSelectedKey(metaData.key);
+   }
  };
```

> Setting state will trigger another render. So in the render method we need generate our table rows with the updated isToggle and onToggle props. Each item needs a unique key, not necessarily associated to our own key, but it works as well. The table renders flat, so keys need to be unique even if they are placed within sections.

Reviewer: ```suggestion
Setting state will trigger another render. In the render method we need to generate our table rows with the updated isToggle and onToggle props. Each item needs a unique key, not necessarily associated to our own key, but it works for this example. The table renders flat, so *all* keys need to be unique even if they are placed within sections.
```

Author: Updated.

> Setting state will trigger another render. In the render method we need generate our table rows with the updated isToggle and onToggle props. Each item needs a unique key, not necessarily associated to our own key, but it works as well. The table renders flat, so keys need to be unique even if they are placed within sections.

Reviewer: ```suggestion
Setting state will trigger another render. In the render method we need to generate our table rows with the updated isToggle and onToggle props. Each item needs a unique key, not necessarily associated to our own key, but it works as well. The table renders flat, so keys need to be unique even if they are placed within sections.
```

import SingleSelectTable from 'terra-table/lib/terra-dev-site/doc/guides/SingleSelectTable?dev-site-example';

# Terra Table - Implementing a Single Select Table

> The terra-table implementation requires controlled state if selections are required. As a result selections are applied at child row generation from the consuming component's state. The following guide show you how to implement that state within a single row selection variant of table.

Reviewer: ```suggestion
The terra-table implementation requires controlled state if selections are required. As a result selections are applied at child row generation from the consuming component's state. The following guide will show you how to implement that state within a single row selection variant of table.
```

Author: Updated.
